src/Workspaces/Core/Portable/Shared/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs

```diff
@@ -336,6 +336,11 @@ private bool HandleNamedTypesWorker(INamedTypeSymbol x, INamedTypeSymbol y, Dict
             {
                 Debug.Assert(GetTypeKind(x) == GetTypeKind(y));
 
+                if (x.IsTupleType)
+                {
+                    return y.IsTupleType && HandleNamedTypesWorker(x.TupleUnderlyingType, y.TupleUnderlyingType, equivalentTypesWithDifferingAssemblies);
```

Comment:
>y.IsTupleType [](start = 27, length = 13)

Do both types have to be tuples in order to be considered equivalent? What if we asked if tuple is equivalent to its underlying type, what is the expected result in that case?